src/jank/clojure/core.jank

```diff
@@ -431,6 +431,92 @@
                  #{ o }#
                );"))
 
+(defn conj!
+  ([]
+   (transient []))
+  ([coll]
+   coll)
+  ([coll x]
+   (native/raw "__value = visit_object
+                (
+                  [=](auto const typed_t, auto const head) -> object_ptr
+                  {
+                    using T = typename decltype(typed_t)::value_type;
+
+                    if constexpr(behavior::persistentable<T>)
+                    { return typed_t->cons_in_place(head); }
+                    else
+                    { throw #{ (ex-info :not-persistentable {:o coll}) }#; }
```

Comment:
In the latest main branch, interpolation has been updated to `~{ foo }` instead of `#{ foo }#`. Merging that in will be necessary first, so that this inline C++ can work.